File: python/palindrome.py
#!/usr/bin/env python

"""
A palindromic number reads the same both ways.
Find the largest palindrome made from the product of two 3-digit numbers.
"""


def main():
    large_pn = 0
    for x in range(100, 999):
        for y in range(100, 999):
            pn = x * y
            if str(pn) == str(pn)[::-1]:
                if pn > large_pn:
                    large_pn = pn

    print large_pn


if __name__ == "__main__":
    main()

File: ruby/palindrome.rb
#!/usr/bin/env ruby


large_pn = 0
1.upto(999) do |x|
  1.upto(999) do |y|
    pn = x * y
    if pn.to_s == pn.to_s.reverse
      large_pn = pn if pn > large_pn
    end
  end
end

puts large_pn